### DISCUSSION TITLE
How does "AF REG" work?

### DISCUSSION THREAD

**Discussion by lxsxl** [PE5PVB/TEF6686_ESP32#338] [General]
I tested the new setting "AF REG". I didn't notice any difference to the old "AF on". TEF switched only within AFs with the same regional variant (= only to AFs with the same PI-code). As there are only a few minutes when the network is splitted in RV I just might have been unlucky. But it would be good to know, so it's easier to test the function.

**Comment by lxsxl** [comment]
I did some further testing: The AF setting has no effect on the colouring of the frequencies in the AF-screen. Receivable regional variants to the tuned frequency are displayed in red colour (only the same regional variant (= same PI-code) may be in green colour). I would expect that in one of the "AF on" modes they will appear in green font. At the same time the AF tuning should also regard those frequencies (= second nibble of PI-code may be different to the PI of the tuned frequency).
To make the settings more understandable they should be named like "Off", "On / REG On" (in this case the AF-tuning shall keep the single regional variant as this is the same label in most car stereos) and "On / REG Off" (in this case the AF-tuning may also switch to a different RV (= second nibble of PI-code may be different).

**Reply by PE5PVB** [reply]
I can make the RV channels a different color when it can be received (so not green). What about blue?

**Comment by PE5PVB** [comment]
When AF-REG is enabled and you are receiving a station with AF-B, AF switching to regional editions is allowed.

**Reply by lxsxl** [reply]
I tried it with the following setting:
1. AF is set to "AF REG"
2. Tuned frequency: 107,1 / DCA4, transmitted AFs i. a. 97,4 and 107,4 (as RV to 107,1)
3. 107,1 with RDS reception so that all AFs are read, then turn antenna a bit so RDS reception is broken -> no tuning to one of the AFs though they are really strong (> 40 dBµV)
I tried this several times, each time with the same result. Tomorrow I can try again with WDR 2, will let you know.

**Comment by lxsxl** [comment]
> I can make the RV channels a different color when it can be received (so not green). What about blue?

Would also be good. But isn't that redundant? If the receivable RVs are also in green colour they can be distinguished by the "R"-indicator in the AF-list.

**Comment by lxsxl** [comment]
I tested again with two different stations similar to the example above. Same results: No AF-tuning at all. Setting was "AF REG"

**Comment by PE5PVB** [comment]
Can you check again? I forgot to add it to the AF scanner.

**Comment by lxsxl** [comment]
I will do tomorrow 👋Am 31.10.2023 um 09:45 schrieb Sjef Verhoeven PE5PVB ***@***.***>:﻿
Can you check again? I forgot to add it to the AF scanner.

**Comment by lxsxl** [comment]
First understanding: On the EON/AF screen the R-AFs are still in red colour. Idk if that is still itended or if it shall be also green (as they will be used as AFs). Switching to a different RV was not yet successful but maybe the quality of reception was too bad. I will try later this evening or at least tomorrow morning.

**Comment by lxsxl** [comment]
I tested again and I still was not able to see one single tuning to an RV-AF. Behaviour is still like descripted in last weeks post.

I will test tomorrow with a different network which is separated only for a few minutes per hour. Those frequencies are much stronger in my place.

But another AF-thing makes me wonder: I also tested with "AF on" for several weaker stations (without RV, only same version). If I tune to a weak frequency and collecting of AF-lists lasts a bit longer then typically there won't be any AF-tuning. This remains even if I scan to EON/AF-screen (and there are definitely much stronger frequencies that are in green colour!) and back. AF-tuning is not possible at all. The impression rises that the program after some seconds of waiting doesn't care for afterwards transmitted (receivable) AFs. Or another idea: If reception quality is too poor, the AF-tuning doesn't work even if there are "green" AFs. Maybe test tomorrow with stronger stations will make it a bit clearer.

**Comment by PE5PVB** [comment]
AF scanning is only triggered when some levels are reached, like noise and multipath. Because it's a single-tuner it won't react like a modern car radio. It's an old school way of using AF. Signal has to go really bad to trigger AF after longer time. This to prevent interruptions in audio.

BTW It's still a DX radio, so most won't ever use AF.

**Comment by lxsxl** [comment]
Tests from today: If TEF is tuned to a station with regional variants (Dx92), there is no tuning to AFs (though strong AFs with different RV are receivable). A few seconds after return to the central PI-codes, it tunes to one of the strongest AFs. Setting was "AF REG".

**Comment by PE5PVB** [comment]
What are the PI's of both stations? Then I can reproduce it.

**Comment by lxsxl** [comment]
I tuned to 101,0 (D492, medium quality but enough to collect AFs), the strongest AF was 100,4 (D992). After some minutes both return to D392 (whers AT-tuning immediately worked).

**Comment by PE5PVB** [comment]
It was a hell of a job, but I think I nailed it. Please test.

**Comment by lxsxl** [comment]
First test in the house: AF-list now shows green AFs even if they are RV. So far so good. AF tuning (actually only tested with central network) shows a small error: If RDS get's lost the font changes to secondary colour. Then automatic tuning to an AF (of course with RDS reception) and the secondary colour remains (at least for PI / PS / PTY; RT get's primary colour when new data are received.

For more I will just do a walk with the dog and let you know about AF tuning on RV...

**Comment by lxsxl** [comment]
My dog was happy that we went for a walk and I was happy that RV-tuning works :-) Well done! But a few details:
1. I think it was nonsense what I wrote in the previous post. Of course the AFs with R-indicator had green font because at that time all had the same central PI.
2. In case of AFs with different PI-codes (RV) they are still displayed in red colour but the AF-tuning works even for them. I started with 107,1 / DCA4 and the radio switched to 97,4 / D6A4 and in a second try to 107,4 / also D6A4. So this was definitely successful.

Actually I have two requests:
1. Rename the setting-options (use AF) from "On" to "On / exclude RV" and "AF REG" to "On / include RV", so it should be clearer what is meant in both cases. Or can I do it myself as a pull request (in which file)?
2. Show also the receivable RV-AFs in green font if the "include RV"-option is chosen.

**Comment by PE5PVB** [comment]
1. I don't think we have enough space for that. It depends on the translator, but my guess is that AF-REG should do it, since it's used widely.
2. It should already do this. I tested it local on a AF-B station I can receive, and I set my transmitter on one of the AF's and change the second nibble from 3 to F, it was always green, as long as AF-REG was selected.

**Reply by lxsxl** [reply]
To 1.: Typical syntax in car stereos is „REG ON“ (= exclude RV [or keep the active RV]) and REG OFF (= include RV). Imho this would be also good and doesn‘t need to be translated.

**Reply by lxsxl** [reply]
To 2.: I will test again tomorrow. Maybe I was too flashed that AF RV worked and didn‘t recognize the green colour 🤷‍♂️

**Comment by PE5PVB** [comment]
OK, So I can change it to status: Off, REG ON and REG OFF?

**Reply by PE5PVB** [reply]
On 2. It will only show you green when AF REG is chosen, otherwise it is red

**Reply by PE5PVB** [reply]
But I can show always green, because the R is in front, you know it's a RV

**Reply by lxsxl** [reply]
Yes, these three status should be good.
And also yes, can be green anyway because of the „R“.

**Comment by lxsxl** [comment]
The RV-AF stay in red colour until AF B method is detected. But it can/should be done without checking  AF method. If PI only differs in second nibble it is an acceptable RV.

As there are only a few minutes to test with the stronger network, I hope that I‘m right: AF tuning also is only done If method B is detected (which is likely in junction with the first part of this post?)
